lib/karafka/params/params.rb

```diff
@@ -55,10 +55,11 @@ def deserialize!
       private
 
       # @param payload [String] Raw data that we want to deserialize using consumer deserializer
+      # option headers [Hash] - optional - Kafka message headers
       # @return [Object] deserialized data
       def deserialize(payload)
         Karafka.monitor.instrument('params.params.deserialize', caller: self) do
-          self['deserializer'].call(payload)
+          self['deserializer'].call(payload, self['headers'])
```

Comment:
I played with 
```ruby
def deserialize(payload, headers = {})
  # ...
end
```

but not sure that it's a good idea. WDYT? I can change it

Reply:
* `headers` sounds good, but I wonder why not pass the whole message (so that we also have access to `topic` for example). 

Idea based on: https://www.confluent.io/blog/put-several-event-types-kafka-topic/ and the 3 strategies defined at the end of the article, especially: `TopicRecordNameStrategy`.

Reply:
good for me, @mensfeld WDYT?

Reply:
I think @paneq has a point here. We could use the whole message. That way someone could use either headers, body or even topic name to build up the messages.

Reply:
That is a breaking change, so would be good to ship before rc1 ;) (I'll wait for that)